typescript/src/policies/schema.ts

```diff
@@ -0,0 +1,266 @@
+import { Address } from "abitype/zod";
+import { z } from "zod";
+
+/**
+ * Enum for EthValueOperator values
+ */
+export const EthValueOperatorEnum = z.enum([">", ">=", "<", "<=", "=="]);
+/**
+ * Type representing the operators that can be used for ETH value comparisons.
+ * These operators determine how transaction values are compared against thresholds.
+ */
+export type EthValueOperator = z.infer<typeof EthValueOperatorEnum>;
+
+/**
+ * Enum for EvmAddressOperator values
+ */
+export const EvmAddressOperatorEnum = z.enum(["in", "not in"]);
+/**
+ * Type representing the operators that can be used for EVM address comparisons.
+ * These operators determine how transaction recipient addresses are evaluated against a list.
+ */
+export type EvmAddressOperator = z.infer<typeof EvmAddressOperatorEnum>;
+
+/**
+ * Enum for SolAddressOperator values
+ */
+export const SolAddressOperatorEnum = z.enum(["in", "not in"]);
+/**
+ * Type representing the operators that can be used for Solana address comparisons.
+ * These operators determine how transaction addresses are evaluated against a list.
+ */
+export type SolAddressOperator = z.infer<typeof SolAddressOperatorEnum>;
+
+/**
+ * Schema for ETH value criterions
+ */
+export const EthValueCriterionSchema = z.object({
+  /** The type of criterion, must be "ethValue" for Ethereum value-based rules. */
+  type: z.literal("ethValue"),
+  /**
+   * The ETH value amount in wei to compare against, as a string.
+   * Must contain only digits.
+   */
+  ethValue: z.string().regex(/^[0-9]+$/),
+  /** The comparison operator to use for evaluating transaction values against the threshold. */
+  operator: EthValueOperatorEnum,
+});
+export type EthValueCriterion = z.infer<typeof EthValueCriterionSchema>;
+
+/**
+ * Schema for EVM address criterions
+ */
+export const EvmAddressCriterionSchema = z.object({
+  /** The type of criterion, must be "evmAddress" for EVM address-based rules. */
+  type: z.literal("evmAddress"),
+  /**
+   * Array of EVM addresses to compare against.
+   * Each address must be a 0x-prefixed 40-character hexadecimal string.
+   * Limited to a maximum of 100 addresses per criterion.
+   */
+  addresses: z.array(Address).max(100),
```

Comment:
wdyt about slapping a `.startsWith("0x", { message: "Must be 0x-prefixed" });` on this bad boy

Reply:
ohh nvm – abitype/zod already handles that, nice